### PR TITLE
fix misc remote build issues

### PR DESCRIPTION
**Pull request by baude**
address problem when multiple -t were sent.  and rework remote build's tarball if a context dir is given other than ".".

Fixes: #6578
Fixes: #6577

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by edsantiago** [comment]
Confirmed: it addresses #6577 and #6578. Thank you!

File: pkg/api/handlers/compat/images_build.go [hunk 1]
```diff
@@ -107,20 +107,24 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
...
 	)
-	if strings.Contains(query.Tag, ":") {
-		tokens := strings.SplitN(query.Tag, ":", 2)
-		name = tokens[0]
-		tag = tokens[1]
+	if len(query.Tag) > 0 {
+		output = query.Tag[0]
 	}
+	if len(query.Tag) > 1 {
+		additionalNames = append(additionalNames, query.Tag[1:]...)
```
Comment (rhatdan) [inline pkg/api/handlers/compat/images_build.go:117]: Nit, this could just be
additionalNames := query.Tag[1:]

**Comment by rhatdan** [comment]
LGTM, Fix Nit, if you find other issues.

File: pkg/domain/infra/tunnel/images.go [hunk 2]
```diff
@@ -288,3 +305,64 @@ func (ir *ImageEngine) Shutdown(_ context.Context) {
...
+			return tw.Close()
+		}
+		if err != nil {
+			return err
+		}
+
+		var body io.Reader = tr
+		name := h.Name
+		data, ok, skip, err := fn(h, tr)
+		logrus.Debugf("Transform %s -> %s: data=%t ok=%t skip=%t err=%v", name, h.Name, data != nil, ok, skip, err)
```
Comment (jwhonce) [inline pkg/domain/infra/tunnel/images.go:332]:
```suggestion
		logrus.Debugf("Transform %q -> %q: data=%t ok=%t skip=%t err=%v", name, h.Name, data != nil, ok, skip, err)
```
small nit for clarity.

File: pkg/domain/infra/tunnel/images.go [hunk 3]
```diff
@@ -288,3 +305,64 @@ func (ir *ImageEngine) Shutdown(_ context.Context) {
...
+	for {
+		h, err := tr.Next()
+		if err == io.EOF {
+			return tw.Close()
+		}
+		if err != nil {
+			return err
+		}
+
+		var body io.Reader = tr
```
Comment (jwhonce) [inline pkg/domain/infra/tunnel/images.go:329]: `tr` doesn't appear to change value in the loop.  Should be moved out of the loop.

File: pkg/domain/infra/tunnel/images.go [hunk 4]
```diff
@@ -288,3 +305,64 @@ func (ir *ImageEngine) Shutdown(_ context.Context) {
...
+		name := h.Name
```
Comment (jwhonce) [inline pkg/domain/infra/tunnel/images.go:330]: From archive/tar documentation:

> For forward compatibility, users that retrieve a Header from Reader.Next, mutate it in some ways, and then pass it back to Writer.WriteHeader should do so by creating a new Header and copying the fields that they are interested in preserving.

Comment (baude) [reply pkg/domain/infra/tunnel/images.go:330]: im seriously hesitant to fiddle with with this as it was taken directly from imagebuilders code, which is presumably tried and true?

File: pkg/api/handlers/compat/images_build.go [hunk 5]
```diff
@@ -107,20 +107,24 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
...
+	}
+	//if strings.Contains(query.Tag, ":") {
```
Comment (mheon) [inline pkg/api/handlers/compat/images_build.go:119]: Can we drop this? Looks like it's no longer needed

**Comment by baude** [comment]
All fixed up with what I am comfortable with ...

**Comment by baude** [comment]
you know the drill ....

**Pull request merged**